packages/tools/src/tools/ScaleOverlayTool.ts

```diff
@@ -114,9 +114,17 @@ class ScaleOverlayTool extends AnnotationDisplayTool {
       )[0] as ScaleOverlayAnnotation;
     }
 
-    // viewportsWithAnnotations stores which viewports have an annotation,
-    // if the viewport does not have an annotation, create a new one
-    if (!viewportsWithAnnotations.includes(viewport.id)) {
+    // viewportsFrameWithAnnotations stores which viewports and FrameOfReference have an annotation,
+    // if the viewport and FrameOfReference does not have an annotation, create a new one
+    const viewportId = viewport.id;
+    const isScaleNotInViewportAndFrameOfReference =
+      !viewportsFrameWithAnnotations.find(
+        (viewport) =>
+          viewport.viewportId == viewportId &&
+          viewport.FrameOfReferenceUID == FrameOfReferenceUID
+      );
+
+    if (isScaleNotInViewportAndFrameOfReference) {
```

Comment:
I'm not sure I follow what is happening here. How is this solving the problem you mentioned? 

I think drag and drop is an app-level concern, so if you just make sure you call setToolEnabled again, it would solve this issue